html/semantics/embedded-content/the-iframe-element/cross-origin-to-whom.window.js

```diff
@@ -0,0 +1,40 @@
+async_test(t => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  frame.src = "support/document-with-embedded-svg.html";
+  const elements = {
+    //"embed": ["getSVGDocument"],
+    "frame": ["contentDocument"],
+    "iframe": ["getSVGDocument", "contentDocument"],
+    "object": ["getSVGDocument", "contentDocument"]
+  };
+  function assert_apis(instance) {
+    const name = instance.localName;
+    let priorPossibleDocument = null;
+    elements[name].forEach(api => {
+      const assertReason = `${name}[${api}]`;
+      let possibleDocument = null;
+      if (api == "getSVGDocument") {
+        possibleDocument = instance[api]();
+      } else {
+        possibleDocument = instance[api];
+      }
+      assert_not_equals(possibleDocument, null, `${name}[${api}]`);
+      // This needs standardizing still
+      // assert_class_string(possibleDocument, "XMLDocument");
+
+      // Ensure getSVGDocument() and contentDocument if both available return the same
+      if (priorPossibleDocument === null) {
+        priorPossibleDocument = possibleDocument;
+      } else {
+        assert_equals(priorPossibleDocument, possibleDocument);
+      }
+    });
+  }
+  frame.onload = t.step_func_done(() => {
+    const instances = Object.keys(elements).map(element => frame.contentDocument.querySelector(element));
+    instances.forEach(instance => assert_apis(instance));
+    document.domain = document.domain;
+    instances.forEach(instance => assert_apis(instance));
+  });
+  document.body.appendChild(frame);
+}, "Test embed/frame/iframe/object nested document APIs for same-origin and cross-origin");
```

Comment:
How is it not? It sets document.domain.

(There's already separate test coverage (and bugs filed) for Chrome/Safari not exposing the right APIs.)

Reply:
Oh, yes, indeed. I think the test name could be clearer about what it's intending to test. I thought "same-origin and cross-origin" would be separate tests, but then same-origin refers to the nested iframe and the further nested SVG, correct? Test names don't need to cover all details, I was just confused from this test name.

Reply:
It refers in particular to the security check explained in HTML and how that applies here. I'm not really sure what a better title would be.

Reply:
It would be clearer to have separate tests and each test has a test name that describes what it's doing.

`Test embed getSVGDocument() same-origin`
`Test embed getSVGDocument() cross-origin by setting document.domain`
etc.

Reply:
I don't think that's worth doing given we already have such tests as I mentioned above.

Reply:
Very well.